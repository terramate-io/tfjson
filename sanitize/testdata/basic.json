{
  "format_version": "0.1",
  "terraform_version": "0.15.1",
  "variables": {
    "foo": {
      "value": "bar"
    },
    "toggle_sensitive": {
      "value": true
    }
  },
  "planned_values": {
    "outputs": {
      "secret": {
        "sensitive": true,
        "value": "7483449866618536328"
      }
    },
    "root_module": {
      "resources": [
        {
          "address": "null_resource.bar",
          "mode": "managed",
          "type": "null_resource",
          "name": "bar",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "4356817825751333663",
            "triggers": {
              "bar": "two",
              "foo": "one"
            }
          }
        },
        {
          "address": "null_resource.baz",
          "mode": "managed",
          "type": "null_resource",
          "name": "baz",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "2737458537890894185",
            "triggers": {
              "foo": "bar"
            }
          }
        },
        {
          "address": "null_resource.foo",
          "mode": "managed",
          "type": "null_resource",
          "name": "foo",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "7483449866618536328",
            "triggers": {
              "bar": "two",
              "foo": "one"
            }
          }
        },
        {
          "address": "null_resource.qux",
          "mode": "managed",
          "type": "null_resource",
          "name": "qux",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "389458145002419915",
            "triggers": {
              "foo": "bar"
            }
          }
        }
      ]
    }
  },
  "resource_changes": [
    {
      "address": "null_resource.bar",
      "mode": "managed",
      "type": "null_resource",
      "name": "bar",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "no-op"
        ],
        "before": {
          "id": "4356817825751333663",
          "triggers": {
            "bar": "two",
            "foo": "one"
          }
        },
        "after": {
          "id": "4356817825751333663",
          "triggers": {
            "bar": "two",
            "foo": "one"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": true
        },
        "after_sensitive": {
          "triggers": true
        }
      }
    },
    {
      "address": "null_resource.baz",
      "mode": "managed",
      "type": "null_resource",
      "name": "baz",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "no-op"
        ],
        "before": {
          "id": "2737458537890894185",
          "triggers": {
            "foo": "bar"
          }
        },
        "after": {
          "id": "2737458537890894185",
          "triggers": {
            "foo": "bar"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": {
            "foo": true
          }
        },
        "after_sensitive": {
          "triggers": {
            "foo": true
          }
        }
      }
    },
    {
      "address": "null_resource.foo",
      "mode": "managed",
      "type": "null_resource",
      "name": "foo",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "no-op"
        ],
        "before": {
          "id": "7483449866618536328",
          "triggers": {
            "bar": "two",
            "foo": "one"
          }
        },
        "after": {
          "id": "7483449866618536328",
          "triggers": {
            "bar": "two",
            "foo": "one"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": {
            "bar": true
          }
        },
        "after_sensitive": {
          "triggers": {
            "bar": true
          }
        }
      }
    },
    {
      "address": "null_resource.qux",
      "mode": "managed",
      "type": "null_resource",
      "name": "qux",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "update"
        ],
        "before": {
          "id": "389458145002419915",
          "triggers": {
            "foo": "bar"
          }
        },
        "after": {
          "id": "389458145002419915",
          "triggers": {
            "foo": "bar"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": {}
        },
        "after_sensitive": {
          "triggers": {
            "foo": true
          }
        }
      }
    }
  ],
  "output_changes": {
    "secret": {
      "actions": [
        "no-op"
      ],
      "before": "7483449866618536328",
      "after": "7483449866618536328",
      "after_unknown": false,
      "before_sensitive": true,
      "after_sensitive": true
    }
  },
  "prior_state": {
    "format_version": "0.1",
    "terraform_version": "0.15.1",
    "values": {
      "outputs": {
        "secret": {
          "sensitive": true,
          "value": "7483449866618536328"
        }
      },
      "root_module": {
        "resources": [
          {
            "address": "data.secret_data_source.current",
            "mode": "data",
            "type": "secret_data_source",
            "name": "current",
            "values": {
              "id": "secret",
              "access_token": "123"
            },
            "provider_name": "secret",
            "schema_version": 0,
            "sensitive_values": {
              "access_token": true
            }
          },
          {
            "address": "data.secret_data_source.object",
            "mode": "data",
            "type": "secret_data_source",
            "name": "object",
            "values": {
              "id": "secret",
              "object": {
                "a": {
                  "a": 1,
                  "b": 2
                },
                "b": {
                  "a": 1,
                  "b": 2
                }
              }
            },
            "provider_name": "secret",
            "schema_version": 0,
            "sensitive_values": {
              "object": true
            }
          },
          {
            "address": "data.secret_data_source.list",
            "mode": "data",
            "type": "secret_data_source",
            "name": "list",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "id": "my_secret",
              "list": [
                "pass1",
                "pass2",
                "pass3"
              ]
            },
            "sensitive_values": {
              "list": true
            }
          },
          {
            "address": "data.secret_data_source.boolean",
            "mode": "data",
            "type": "secret_data_source",
            "name": "boolean",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "id": "my_secret",
              "boolean": true
            },
            "sensitive_values": {
              "boolean": true
            }
          },
          {
            "address": "data.secret_data_source.nested",
            "mode": "data",
            "type": "secret_data_source",
            "name": "nested",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "nested": {
                "some": "value"
              }
            },
            "sensitive_values": {
              "nested": {
                "some": true
              }
            }
          },
          {
            "address": "data.secret_data_source.nested_list",
            "mode": "data",
            "type": "secret_data_source",
            "name": "nested_list",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "nested_list": [
                "not secret",
                "secret",
                "not secret"
              ]
            },
            "sensitive_values": {
              "nested_list": [
                false,
                true,
                false
              ]
            }
          },
          {
            "address": "data.secret_data_source.nested_list2",
            "mode": "data",
            "type": "secret_data_source",
            "name": "nested_list",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "nested_list2": [
                {
                  "some": "not secret"
                },
                {
                  "some": "secret"
                },
                "not secret"
              ]
            },
            "sensitive_values": {
              "nested_list2": [
                {
                  "some": false
                },
                {
                  "some": true
                },
                false
              ]
            }
          },
          {
            "address": "null_resource.bar",
            "mode": "managed",
            "type": "null_resource",
            "name": "bar",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "4356817825751333663",
              "triggers": {
                "bar": "two",
                "foo": "one"
              }
            }
          },
          {
            "address": "null_resource.baz",
            "mode": "managed",
            "type": "null_resource",
            "name": "baz",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "2737458537890894185",
              "triggers": {
                "foo": "bar"
              }
            }
          },
          {
            "address": "null_resource.foo",
            "mode": "managed",
            "type": "null_resource",
            "name": "foo",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "7483449866618536328",
              "triggers": {
                "bar": "two",
                "foo": "one"
              }
            }
          },
          {
            "address": "null_resource.qux",
            "mode": "managed",
            "type": "null_resource",
            "name": "qux",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "389458145002419915",
              "triggers": {
                "foo": "bar"
              }
            }
          }
        ]
      }
    }
  },
  "configuration": {
    "root_module": {
      "outputs": {
        "secret": {
          "sensitive": true,
          "expression": {
            "references": [
              "null_resource.foo"
            ]
          }
        }
      },
      "resources": [
        {
          "address": "null_resource.bar",
          "mode": "managed",
          "type": "null_resource",
          "name": "bar",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {}
          },
          "schema_version": 0
        },
        {
          "address": "null_resource.baz",
          "mode": "managed",
          "type": "null_resource",
          "name": "baz",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {
              "references": [
                "var.foo"
              ]
            }
          },
          "schema_version": 0
        },
        {
          "address": "null_resource.foo",
          "mode": "managed",
          "type": "null_resource",
          "name": "foo",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {}
          },
          "schema_version": 0
        },
        {
          "address": "null_resource.qux",
          "mode": "managed",
          "type": "null_resource",
          "name": "qux",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {
              "references": [
                "var.toggle_sensitive"
              ]
            }
          },
          "schema_version": 0
        }
      ],
      "variables": {
        "foo": {
          "default": "bar",
          "sensitive": true
        },
        "toggle_sensitive": {}
      }
    }
  }
}