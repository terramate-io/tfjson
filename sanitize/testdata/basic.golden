{
  "format_version": "0.1",
  "terraform_version": "0.15.1",
  "variables": {
    "foo": {
      "value": "REDACTED_SENSITIVE"
    },
    "toggle_sensitive": {
      "value": true
    }
  },
  "planned_values": {
    "outputs": {
      "secret": {
        "sensitive": true,
        "value": "REDACTED_SENSITIVE"
      }
    },
    "root_module": {
      "resources": [
        {
          "address": "null_resource.bar",
          "mode": "managed",
          "type": "null_resource",
          "name": "bar",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "4356817825751333663",
            "triggers": "REDACTED_SENSITIVE"
          }
        },
        {
          "address": "null_resource.baz",
          "mode": "managed",
          "type": "null_resource",
          "name": "baz",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "2737458537890894185",
            "triggers": {
              "foo": "REDACTED_SENSITIVE"
            }
          }
        },
        {
          "address": "null_resource.foo",
          "mode": "managed",
          "type": "null_resource",
          "name": "foo",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "7483449866618536328",
            "triggers": {
              "bar": "REDACTED_SENSITIVE",
              "foo": "one"
            }
          }
        },
        {
          "address": "null_resource.qux",
          "mode": "managed",
          "type": "null_resource",
          "name": "qux",
          "provider_name": "registry.terraform.io/hashicorp/null",
          "schema_version": 0,
          "values": {
            "id": "389458145002419915",
            "triggers": {
              "foo": "REDACTED_SENSITIVE"
            }
          }
        }
      ]
    }
  },
  "resource_changes": [
    {
      "address": "null_resource.bar",
      "mode": "managed",
      "type": "null_resource",
      "name": "bar",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "no-op"
        ],
        "before": {
          "id": "4356817825751333663",
          "triggers": "REDACTED_SENSITIVE"
        },
        "after": {
          "id": "4356817825751333663",
          "triggers": "REDACTED_SENSITIVE"
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": true
        },
        "after_sensitive": {
          "triggers": true
        }
      }
    },
    {
      "address": "null_resource.baz",
      "mode": "managed",
      "type": "null_resource",
      "name": "baz",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "no-op"
        ],
        "before": {
          "id": "2737458537890894185",
          "triggers": {
            "foo": "REDACTED_SENSITIVE"
          }
        },
        "after": {
          "id": "2737458537890894185",
          "triggers": {
            "foo": "REDACTED_SENSITIVE"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": {
            "foo": true
          }
        },
        "after_sensitive": {
          "triggers": {
            "foo": true
          }
        }
      }
    },
    {
      "address": "null_resource.foo",
      "mode": "managed",
      "type": "null_resource",
      "name": "foo",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "no-op"
        ],
        "before": {
          "id": "7483449866618536328",
          "triggers": {
            "bar": "REDACTED_SENSITIVE",
            "foo": "one"
          }
        },
        "after": {
          "id": "7483449866618536328",
          "triggers": {
            "bar": "REDACTED_SENSITIVE",
            "foo": "one"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": {
            "bar": true
          }
        },
        "after_sensitive": {
          "triggers": {
            "bar": true
          }
        }
      }
    },
    {
      "address": "null_resource.qux",
      "mode": "managed",
      "type": "null_resource",
      "name": "qux",
      "provider_name": "registry.terraform.io/hashicorp/null",
      "change": {
        "actions": [
          "update"
        ],
        "before": {
          "id": "389458145002419915",
          "triggers": {
            "foo": "bar"
          }
        },
        "after": {
          "id": "389458145002419915",
          "triggers": {
            "foo": "REDACTED_SENSITIVE"
          }
        },
        "after_unknown": {},
        "before_sensitive": {
          "triggers": {}
        },
        "after_sensitive": {
          "triggers": {
            "foo": true
          }
        }
      }
    }
  ],
  "output_changes": {
    "secret": {
      "actions": [
        "no-op"
      ],
      "before": "REDACTED_SENSITIVE",
      "after": "REDACTED_SENSITIVE",
      "after_unknown": false,
      "before_sensitive": true,
      "after_sensitive": true
    }
  },
  "prior_state": {
    "format_version": "0.1",
    "terraform_version": "0.15.1",
    "values": {
      "outputs": {
        "secret": {
          "sensitive": true,
          "value": "REDACTED_SENSITIVE"
        }
      },
      "root_module": {
        "resources": [
          {
            "address": "data.secret_data_source.current",
            "mode": "data",
            "type": "secret_data_source",
            "name": "current",
            "provider_name": "secret",
            "schema_version": 0,
            "values": {
              "access_token": "REDACTED_SENSITIVE",
              "id": "secret"
            },
            "sensitive_values": {
              "access_token": true
            }
          },
          {
            "address": "data.secret_data_source.object",
            "mode": "data",
            "type": "secret_data_source",
            "name": "object",
            "provider_name": "secret",
            "schema_version": 0,
            "values": {
              "id": "secret",
              "object": "REDACTED_SENSITIVE"
            },
            "sensitive_values": {
              "object": true
            }
          },
          {
            "address": "data.secret_data_source.list",
            "mode": "data",
            "type": "secret_data_source",
            "name": "list",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "id": "my_secret",
              "list": "REDACTED_SENSITIVE"
            },
            "sensitive_values": {
              "list": true
            }
          },
          {
            "address": "data.secret_data_source.boolean",
            "mode": "data",
            "type": "secret_data_source",
            "name": "boolean",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "boolean": "REDACTED_SENSITIVE",
              "id": "my_secret"
            },
            "sensitive_values": {
              "boolean": true
            }
          },
          {
            "address": "data.secret_data_source.nested",
            "mode": "data",
            "type": "secret_data_source",
            "name": "nested",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "nested": {
                "some": "REDACTED_SENSITIVE"
              }
            },
            "sensitive_values": {
              "nested": {
                "some": true
              }
            }
          },
          {
            "address": "data.secret_data_source.nested_list",
            "mode": "data",
            "type": "secret_data_source",
            "name": "nested_list",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "nested_list": [
                "not secret",
                "REDACTED_SENSITIVE",
                "not secret"
              ]
            },
            "sensitive_values": {
              "nested_list": [
                false,
                true,
                false
              ]
            }
          },
          {
            "address": "data.secret_data_source.nested_list2",
            "mode": "data",
            "type": "secret_data_source",
            "name": "nested_list",
            "provider_name": "my_secret",
            "schema_version": 0,
            "values": {
              "nested_list2": [
                {
                  "some": "not secret"
                },
                {
                  "some": "REDACTED_SENSITIVE"
                },
                "not secret"
              ]
            },
            "sensitive_values": {
              "nested_list2": [
                {
                  "some": false
                },
                {
                  "some": true
                },
                false
              ]
            }
          },
          {
            "address": "null_resource.bar",
            "mode": "managed",
            "type": "null_resource",
            "name": "bar",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "4356817825751333663",
              "triggers": "REDACTED_SENSITIVE"
            }
          },
          {
            "address": "null_resource.baz",
            "mode": "managed",
            "type": "null_resource",
            "name": "baz",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "2737458537890894185",
              "triggers": {
                "foo": "REDACTED_SENSITIVE"
              }
            }
          },
          {
            "address": "null_resource.foo",
            "mode": "managed",
            "type": "null_resource",
            "name": "foo",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "7483449866618536328",
              "triggers": {
                "bar": "REDACTED_SENSITIVE",
                "foo": "one"
              }
            }
          },
          {
            "address": "null_resource.qux",
            "mode": "managed",
            "type": "null_resource",
            "name": "qux",
            "provider_name": "registry.terraform.io/hashicorp/null",
            "schema_version": 0,
            "values": {
              "id": "389458145002419915",
              "triggers": {
                "foo": "bar"
              }
            }
          }
        ]
      }
    }
  },
  "configuration": {
    "root_module": {
      "outputs": {
        "secret": {
          "sensitive": true,
          "expression": {
            "references": [
              "null_resource.foo"
            ]
          }
        }
      },
      "resources": [
        {
          "address": "null_resource.bar",
          "mode": "managed",
          "type": "null_resource",
          "name": "bar",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {}
          },
          "schema_version": 0
        },
        {
          "address": "null_resource.baz",
          "mode": "managed",
          "type": "null_resource",
          "name": "baz",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {
              "references": [
                "var.foo"
              ]
            }
          },
          "schema_version": 0
        },
        {
          "address": "null_resource.foo",
          "mode": "managed",
          "type": "null_resource",
          "name": "foo",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {}
          },
          "schema_version": 0
        },
        {
          "address": "null_resource.qux",
          "mode": "managed",
          "type": "null_resource",
          "name": "qux",
          "provider_config_key": "null",
          "expressions": {
            "triggers": {
              "references": [
                "var.toggle_sensitive"
              ]
            }
          },
          "schema_version": 0
        }
      ],
      "variables": {
        "foo": {
          "default": "bar",
          "sensitive": true
        },
        "toggle_sensitive": {}
      }
    }
  }
}